internal/controller/components/reconciler.go

```diff
@@ -156,6 +185,44 @@ func Reconcile(
 			continue
 		}
 
+		if needsPullSecret {
```

Comment:
style: might be a good idea to extract this code block into a dedicated method/function